k256/src/arithmetic.rs

```diff
@@ -12,11 +12,19 @@ pub(crate) mod scalar;
 mod dev;
 
 pub use field::FieldElement;
-pub use mul::lincomb;
 
 use self::{affine::AffinePoint, projective::ProjectivePoint, scalar::Scalar};
 use crate::Secp256k1;
-use elliptic_curve::CurveArithmetic;
+use elliptic_curve::{group, CurveArithmetic};
+
+/// Linear combination.
+///
+/// This trait enables providing an optimized implementation of
+/// linear combinations (e.g. Shamir's Trick).
+pub trait LinearCombination<PointsAndScalars: AsRef<[(Self, Self::Scalar)]>>: group::Curve {
+    /// Calculates `x1 * k1 + ... + xn * kn`.
+    fn linear_combination(points_and_scalars: PointsAndScalars) -> Self;
+}
```

Comment:
I opened a PR which adds a similar trait upstream to `elliptic-curve` here: https://github.com/RustCrypto/traits/pull/1405

Reply:
Ok I adapted the code to work with that